subgraphs/_reference_/src/sdk/protocols/bridge/tokens.ts

```diff
@@ -82,9 +82,9 @@ export class TokenManager {
 
   getOrCreateCrosschainToken(
     chainID: BigInt,
-    address: Address,
+    address: Bytes,
     type: string,
-    token: Address
+    token: Bytes
```

Comment:
Similar thing here. This is a breaking change to the API. There is a chance it works because Address inherits from Bytes, so AS might cast automatically and everything might work.

But instead of changing public contracts, I'd suggest to add new functionality on top instead.

In this case, a new function like:
`getOrCreateNonEVMCrosschainToken(chainID: BigInt, address: Bytes, type: string, token: Address): CrosschainToken`

Reply:
Exactly, I want to take advantage of the fact Address inheriting from Bytes, and it should not break current code. I think the current change has the advantage that the same interfaces/methods will work for both EVM and non-EVM chains. If we use different methods for EVM and non-EVM transfer, we will need a way to lookup non-EVM chains or detect non-EVM address.

Reply:
OK, I realized I need to treat non-EVM address differently anyway with `Address.fromUTF8(non-EVM Address)` instead of `Address.fromString(EVM address)`, but I do not fully grasp why it is better to have two different methods if one can do.

Reply:
> If we use different methods for EVM and non-EVM transfer, we will need a way to lookup non-EVM chains or detect non-EVM address.

Fair point. The question then is if we want the SDK to handle that magically or to pass that onto the client, since they are closer to the given protocol and might know better (but gives them additional overhead). I have my preference, but no strong feelings :D 

> but I do not fully grasp why it is better to have two different methods if one can do.

I personally prefer to be explicit about the purpose of each method. Specially if the alternative involves modifying its public signature. Also by always defaulting to break everything down that much it makes it harder to end up with a big function/class with too many responsibilities.

If it will be backwards compatible, and being that the purpose of the function is essentially the same, it might not be that big of a deal, though. I'll leave it up to you, your call :)

Reply:
Your point makes more sense. Do you think it is needed to create a separate `getOrCreateNonEVMCrosschainToken` method for token? The only difference is to change argument to `address: Bytes` from `address: Address`.